elastic/eui
# [EuiPortal] Allow `insert` prop to be configured via `EuiProvider.componentDefaults`

## Summary

> ⚠️ This PR contains semi-messy git history (due to me not wanting to totally nuke @y1j2x34's commits/contributions) - since this PR is on the smaller side, I'd recommend simply reviewing the [final files changed diff](https://github.com/elastic/eui/pull/6941/files) instead.

closes https://github.com/elastic/eui/issues/6882 (although this work won't be available until this feature branch is merged into main and released)

Example usage:

```tsx
<EuiProvider
  componentDefaults={{
    EuiPortal: { insert: { sibling: document.querySelector('.foo'), position: 'after' } },
  }}
>
  <App />
</EuiProvider>
```

See https://eui.elastic.co/pr_6923/#/utilities/provider#component-defaults for more documentation.

## QA

- Run `yarn test-cypress-dev` locally
- Open Chrome component testing, select `portal.spec.tsx`
- Confirm all / new `EuiProvider.componentDefaults` tests pass
- Optionally inspect the DOM output of the two tests to confirm that 2 portal nodes rendered in the expected positions, based on the passed code

### General checklist
